Compilation issue, latest code

I used the latest code to compile it in RPI. Before that, it was good,Before that, such as label v3.7
/usr/bin/ld: libbaresip.so.11.7.0: undefined reference to `__atomic_load_8'
/usr/bin/ld: libbaresip.so.11.7.0: undefined reference to `__atomic_store_8'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/baresip_exe.dir/build.make:108: baresip] Error 1
make[1]: *** [CMakeFiles/Makefile2:1188: CMakeFiles/baresip_exe.dir/all] Error 2
make: *** [Makefile:171: all] Error 2

> You can try:
> 
> `-DCMAKE_REQUIRED_LIBRARIES="-latomic"`

Tested, it's not working. I'll see what the reason is later


> Can you try this PR: [baresip/re#1029](https://github.com/baresip/re/pull/1029)
[ 22%] Building C object CMakeFiles/baresip_exe.dir/src/main.c.o
[ 23%] Linking C executable baresip
/usr/bin/ld: libbaresip.so.11.7.0: undefined reference to `__atomic_load_8'
/usr/bin/ld: libbaresip.so.11.7.0: undefined reference to `__atomic_store_8'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/baresip_exe.dir/build.make:108: baresip] Error 1
make[1]: *** [CMakeFiles/Makefile2:1188: CMakeFiles/baresip_exe.dir/all] Error 2
make: *** [Makefile:171: all] Error 2

Tested, it's not working.
step:
1. re update branch and cmake make ( git checkout -b origin/cmake_atomic_detection)
2. baresip rebuild cmake and make

other tested
pi@raspberrypi:~/code/baresip/baresip/build $ ls
a.out           CMakeFiles           compile_commands.json  CPackSourceConfig.cmake  libbaresip.so     libbaresip.so.11.7.0  modules    test
CMakeCache.txt  cmake_install.cmake  CPackConfig.cmake      libbaresip.pc            libbaresip.so.11  Makefile              packaging  webrtc
pi@raspberrypi:~/code/baresip/baresip/build $ 
pi@raspberrypi:~/code/baresip/baresip/build $ **ld libbaresip.so.11**
ld: warning: cannot find entry symbol _start; defaulting to 00010154
ld: libbaresip.so.11: undefined reference to `__atomic_load_8'
ld: libbaresip.so.11: undefined reference to `__atomic_store_8'
pi@raspberrypi:~/code/baresip/baresip/build $ **ld libbaresip.so.11 -latomic**
ld: warning: cannot find entry symbol _start; defaulting to 00010164

-- Performing Test atomic_test
-- Performing Test atomic_test - Failed
-- Performing Test atomic_test_lib
-- Performing Test atomic_test_lib - Failed
CMake Error at cmake/re-config.cmake:288 (message):

pi@raspberrypi:~/code/baresip/re/build $ cat /home/pi/code/baresip/re/build/CMakeFiles/CMakeError.log
![image](https://github.com/baresip/baresip/assets/38849089/7f6f5561-827f-41a6-b821-5a6ceb50b683)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -242,21 +242,10 @@ endif()
 # Main target library
 #
 
-set(LINKLIBS ${RE_LIBRARIES} Threads::Threads)
-
-if(ZLIB_FOUND)
-  list(APPEND LINKLIBS ZLIB::ZLIB)
-endif()
-
-if(USE_OPENSSL)
-  list(APPEND LINKLIBS OpenSSL::SSL OpenSSL::Crypto)
-endif()
+set(LINKLIBS ${RE_LIBRARIES} ${RE_LIBS})
 
 if(WIN32)
-  list(APPEND LINKLIBS ws2_32 iphlpapi winmm gdi32 crypt32 strmiids
-      ole32 oleaut32 qwave dbghelp)
-else()
-  list(APPEND LINKLIBS -lm ${RESOLV_LIBRARY})
+  list(APPEND LINKLIBS winmm gdi32 crypt32 strmiids ole32 oleaut32)
 endif()
 
 if(STATIC)

